### PR TITLE
Ags multiple angles

### PR DESCRIPTION
**Pull request by cquiroz**
With this PR we can do the AGS calculation across many positions, typically many angles for the flip and unbounded option. Previously we could pass a single angle

### CODE REVIEW

File: modules/ags/src/main/scala/lucuma/ags/Ags.scala [hunk 1]
```diff
@@ -170,21 +180,23 @@ object Ags {
...
     // use constraints to calculate all guide speeds
     val bc          = constraintsFor(guideSpeeds)
 
     in =>
-      in.filter(c => c.gBrightness.exists(g => bc.exists(_.contains(Band.Gaia, g))))
-        .map { gsc =>
+      (in.filter(c => c.gBrightness.exists(g => bc.exists(_.contains(Band.Gaia, g)))),
+       Stream.emits[F, AgsPosition](positions.toList)
+      )
+        .mapN { (gsc, position) =>
```
Comment (toddburnside) [inline modules/ags/src/main/scala/lucuma/ags/Ags.scala:199]: Ah. Here is the cartesian product you were asking about in slack. :)

**Review by toddburnside — APPROVED** [review]
LGTM

File: modules/ags/src/main/scala/lucuma/ags/Ags.scala (hunk 1 above)
Comment (cquiroz) [reply modules/ags/src/main/scala/lucuma/ags/Ags.scala:199]: Yes indeed

File: modules/tests/jvm/src/main/scala/lucuma/catalog/AgsSelectionStreamApp.scala [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                  GmosNorthFpu.LongSlit_1_00.asLeft.some,
+                  PortDisposition.Side
+                )
+              )
+          )
+          .compile
+          .toList
+      )
+      .flatTap(x => IO.println(x.length))
+      // .flatMap(x => IO.println(x.head))
```
Comment (rpiaggio) [inline modules/tests/jvm/src/main/scala/lucuma/catalog/AgsSelectionStreamApp.scala:81]: Commented code. Also uncommented code that looks like debug code.

Comment (rpiaggio) [reply modules/tests/jvm/src/main/scala/lucuma/catalog/AgsSelectionStreamApp.scala:81]: Ooops, just realized this is a test app, so the idea is to print things out.

**Review by rpiaggio — APPROVED** [review]
LGTM

**Pull request merged**